### PR TITLE
Feature/api endpoints

### PR DESCRIPTION
**Pull request by RileyManda**
The following milestone implementations has been completed according to stipulated [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_api_endpoints.md): Adding API Endpoints : 🌱 

# Implementations: 🧰  🔧 

### Api- End Points 🤝 

- Created an API endpoint to list all posts for a user.
- Created an API endpoint to list all comments for a user's post.
- Created an API endpoint to add a comment to a post.
- API endpoints receive JSON and respond JSON as well.

 In addition  following general requirements have been fulfilled:
 
### General requirements

> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).

Please kindly review this implementation.

Thank you 🙏🏻

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -4,6 +4,10 @@ def index
     @posts = Post.all
     @user = User.find(params[:user_id])
     @posts = @user.posts
+    respond_to do |format|
+      format.html
+      format.json { render json: @posts }
+    end
   end
```
Comment (LionRouge1) [inline app/controllers/posts_controller.rb]: - [ ] The API endpoint to list all the posts created by a user isn't functioning as expected due to Devise authentication(see image). Kindly check and make sure the response is a JSON.
Screenshot from 2023-10-12 19-25-15

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -2,17 +2,27 @@ class CommentsController < ApplicationController
   load_and_authorize_resource
   before_action :find_post
 
+  def index
+    @comments = @post.comments
+    respond_to do |format|
+      format.html
+      format.json { render json: @comments }
+    end
```
Comment (LionRouge1) [inline app/controllers/comments_controller.rb:10]: - [ ] The API endpoint to list all the posts' comments isn't functioning as expected due to Devise authentication(see image). Kindly check and make sure the response is a JSON.
Screenshot from 2023-10-12 19-51-24

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -2,17 +2,27 @@ class CommentsController < ApplicationController
...
-    else
-      flash[:error] = 'Error creating the comment.'
+    respond_to do |format|
+      if @comment.save
+        format.html { redirect_to user_post_path(@comment.post.author_id, @comment.post.id) }
+        format.json { render json: @comment, status: :created }
+      else
+        format.html { render :new }
+        format.json { render json: @comment.errors, status: :unprocessable_entity }
+      end
```
Comment (LionRouge1) [inline app/controllers/comments_controller.rb:24]: - [ ] Same for the API endpoint for adding a new comment to a post

**Review by LionRouge1 — CHANGES_REQUESTED** [review]
Hi @RileyManda & @H-Ramalan, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

[OPTIONAL]
- [ ] It's a good practice to add `/api` to the URL or `api/v1` with the version number in order to identify it as an API-based URL and it also helps to separate the API's controllers from the other controllers. You can archive this by making use of the rails controller **Namespaces and routing**, you can read more about [namespace here](https://guides.rubyonrails.org/routing.html#controller-namespaces-and-routing)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by RileyManda** [comment]
The following itertions have been implemented to the project: Api endpoints:

Iterations: 🔨 

- Fixed issue related to displaying comments and posts data

- added serializers

- Updated routes to include api path: api/v1/

Screen Shot 2023-10-13 at 11 40 48 AM
Screen Shot 2023-10-13 at 11 43 46 AM


In addition the following general requirements have been fulfilled:
 
### General requirements




> Made sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
> Made sure that I have used correct flow [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
> Made sure that I have documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
> Followed list of [best practices for Ruby](https://github.com/microverseinc/curriculum-ruby/blob/main/articles/ruby_best_practices.md).


Please kindly review this implementation.

Thank you 🙏🏻

**Review by addod19 — APPROVED** [review]
## Hi @RileyManda and @H-Ramalan, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

### Highlights
- [ ] Great job fixing the required changes as requested from the previous review :rocket: 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**